In [9]:
#ファイルの読み込み、中身を配列に格納、集合用の辞書を用意
col1 = open("col1.txt","r")
content = col1.read().splitlines()     
set_col1 = {}

#1行ずつ読み出し、辞書に含まれていなければ出現回数を記録
for name in content:
    if name not in set_col1:
        set_col1[name] = content.count(name)

#作成した辞書を出現回数で降順にソート
sorted_set_col1 = sorted(set_col1.items(),reverse=True,key=lambda x:x[1])

#1タプルずつ表示
for name,count in sorted_set_col1:
    print(count,name)

col1.close()

118 James
111 William
108 John
108 Robert
92 Mary
75 Charles
74 Michael
73 Elizabeth
70 Joseph
60 Margaret
58 George
58 Thomas
57 David
51 Richard
45 Helen
43 Frank
43 Christopher
41 Anna
40 Edward
39 Ruth
38 Patricia
37 Matthew
36 Dorothy
35 Emma
32 Barbara
31 Daniel
31 Joshua
26 Sarah
26 Linda
26 Jennifer
26 Emily
25 Jessica
25 Jacob
24 Mildred
24 Betty
24 Susan
23 Henry
23 Ashley
22 Nancy
21 Andrew
20 Florence
20 Marie
20 Donald
20 Amanda
19 Samantha
18 Karen
18 Lisa
18 Melissa
18 Madison
18 Olivia
17 Stephanie
17 Abigail
16 Ethel
16 Sandra
16 Mark
15 Frances
15 Carol
15 Angela
15 Michelle
15 Heather
15 Ethan
15 Isabella
14 Shirley
14 Kimberly
14 Amy
14 Ava
13 Virginia
13 Deborah
13 Brian
13 Jason
13 Nicole
13 Hannah
13 Sophia
12 Minnie
12 Bertha
12 Donna
11 Cynthia
10 Alice
10 Doris
10 Ronald
10 Brittany
10 Nicholas
10 Mia
10 Noah
9 Joan
9 Debra
9 Tyler
8 Ida
8 Clara
8 Judith
8 Taylor
8 Alexis
8 Alexander
8 Mason
7 Harry
7 Sharon
7 Steven
7 Tammy
7 Brandon
7 Liam
6 Anthony
5 Annie


--UNIXコマンドによる確認画面--

・uniq 連続して同じ値が出た時に1つにまとめることができる(sortを事前に実行する
・uniq -c(op) まとめた要素の数を記録できる
・sort -r 逆順にソート
・sort -k keyを指定

shota@LAPTOP-7H1S837B:~/py_venvs/nlp100/100knock2024/shota/chapter02$ cut -f 1 popular-names.txt | sort | uniq -c | sort -r -k 1
    118 James
    111 William
    108 Robert
    108 John
     92 Mary
     75 Charles
     74 Michael
     73 Elizabeth
     70 Joseph
     60 Margaret
     58 Thomas
     58 George
     57 David
     51 Richard
     45 Helen
     43 Frank
     43 Christopher
     41 Anna
     40 Edward
     39 Ruth
     38 Patricia
     37 Matthew
     36 Dorothy
     35 Emma
     32 Barbara
     31 Joshua
     31 Daniel
     26 Sarah
     26 Linda
     26 Jennifer
     26 Emily
     25 Jessica
     25 Jacob
     24 Susan
     24 Mildred
     24 Betty
     23 Henry
     23 Ashley
     22 Nancy
     21 Andrew
     20 Marie
     20 Florence
     20 Donald
     20 Amanda
     19 Samantha
     18 Olivia
     18 Melissa
     18 Madison
     18 Lisa
     18 Karen
     17 Stephanie
     17 Abigail
     16 Sandra
     16 Mark
     16 Ethel
     15 Michelle
     15 Isabella
     15 Heather
     15 Frances
     15 Ethan
     15 Carol
     15 Angela
     14 Shirley
     14 Kimberly
     14 Ava
     14 Amy
     13 Virginia
     13 Sophia
     13 Nicole
     13 Jason
     13 Hannah
     13 Deborah
     13 Brian
     12 Minnie
     12 Donna
     12 Bertha
     11 Cynthia
     10 Ronald
     10 Noah
     10 Nicholas
     10 Mia
     10 Doris
     10 Brittany
     10 Alice
      9 Tyler
      9 Joan
      9 Debra
      8 Taylor
      8 Mason
      8 Judith
      8 Ida
      8 Clara
      8 Alexis
      8 Alexander
      7 Tammy
      7 Steven
      7 Sharon
      7 Liam
      7 Harry
      7 Brandon
      6 Anthony
      5 Jeffrey
      5 Jayden
      5 Gary
      5 Charlotte
      5 Annie
      4 Lillian
      4 Kathleen
      4 Justin
      4 Chloe
      4 Benjamin
      4 Austin
      3 Megan
      3 Harper
      3 Evelyn
      3 Elijah
      3 Aiden
      2 Rebecca
      2 Oliver
      2 Logan
      2 Lauren
      2 Larry
      2 Bessie
      2 Amelia
      1 Walter
      1 Tracy
      1 Scott
      1 Rachel
      1 Pamela
      1 Lucas
      1 Lori
      1 Laura
      1 Kelly
      1 Julie
      1 Crystal
      1 Carolyn
